pkg/genericcache/genericcache.go

```diff
@@ -0,0 +1,89 @@
+package genericcache
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/portworx/sched-ops/k8s/apiextensions"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+type genericCache struct {
+	crdKindToPluralMap map[string]string
+}
+
+// GenericCache is a generic structure which will implement mostly non-k8s resources related information.
+// Individual members get and set need to be provided by the interface.
+type GenericCache interface {
+	// GetCRDKindToPluralMap returns the crdKindToPluralMap from generic cache singleton.
+	GetCRDKindToPluralMap() map[string]string
+
+	// SetCRDKindToPluralMap will set a new CRD's plural in the map
+	SetCRDKindToPluralMap(kind, plural string)
+}
+
+var (
+	genericCacheLock sync.Mutex
```

Comment:
move these into the `genericCache` struct instead of global variable.

Reply:
As discussed kept this for accessing the global variable and also introduced a lock inside struct to avoid accessing the map at same time.